tests/scripts/generate_test_code.py

```diff
@@ -0,0 +1,837 @@
+        snippets['test_common_helpers'] = help_f.read()
+        snippets['test_platform_file'] = platform_file
+        snippets['platform_code'] = platform_f.read().replace(
+            'DATA_FILE', out_data_file.replace('\\', '\\\\')) # escape '\'
+
+    # Function code
+    with FileWrapper(funcs_file) as funcs_f, FileWrapper(data_file) as \
+            data_f, open(out_data_file, 'w') as out_data_f:
+        suite_deps, dispatch_code, func_code, func_info = parse_functions(
+            funcs_f)
+        snippets['functions_code'] = func_code
+        snippets['dispatch_code'] = dispatch_code
+        dep_check_code, expression_code = gen_from_test_data(
+            data_f, out_data_f, func_info, suite_deps)
+        snippets['dep_check_code'] = dep_check_code
+        snippets['expression_code'] = expression_code
+
+    snippets['test_file'] = c_file
+    snippets['test_main_file'] = template_file
+    snippets['test_case_file'] = funcs_file
+    snippets['test_case_data_file'] = data_file
+    # Read Template
+    # Add functions
+    #
+    with open(template_file, 'r') as template_f, open(c_file, 'w') as c_f:
+        line_no = 1
+        for line in template_f.readlines():
+            # Update line number. +1 as #line directive sets next line number
+            snippets['line_no'] = line_no + 1
+            code = line.format(**snippets)
+            c_f.write(code)
+            line_no += 1
+
+
+def check_cmd():
+    """
+    Command line parser.
+
+    :return:
+    """
+    parser = argparse.ArgumentParser(
+        description='Dynamically generate test suite code.')
+
+    parser.add_argument("-f", "--functions-file",
+                        dest="funcs_file",
+                        help="Functions file",
+                        metavar="FUNCTIONS_FILE",
+                        required=True)
+
+    parser.add_argument("-d", "--data-file",
+                        dest="data_file",
+                        help="Data file",
+                        metavar="DATA_FILE",
+                        required=True)
+
+    parser.add_argument("-t", "--template-file",
+                        dest="template_file",
+                        help="Template file",
+                        metavar="TEMPLATE_FILE",
+                        required=True)
+
+    parser.add_argument("-s", "--suites-dir",
+                        dest="suites_dir",
+                        help="Suites dir",
+                        metavar="SUITES_DIR",
+                        required=True)
+
+    parser.add_argument("--helpers-file",
+                        dest="helpers_file",
+                        help="Helpers file",
+                        metavar="HELPERS_FILE",
+                        required=True)
+
+    parser.add_argument("-p", "--platform-file",
+                        dest="platform_file",
+                        help="Platform code file",
+                        metavar="PLATFORM_FILE",
+                        required=True)
```

Comment:
I think that the help message of these options is too limited, I suggest expanding the help messages.

Reply:
Documentation is elaborated in the module description at the top of this file. That description appears in the pydoc viewed via python console. Option description is limited here to avoid cluter in `--help` output.

Reply:
`--help` doesn't need to be extremely detailed, but it could use improvements. Describing `--help-file` as “Help file” is not helpful.

Reply:
It is described as "Help**ers** file". What helpers file means is described in module documentation (in coming up commits).

Reply:
You don't need to open the code. For python module docstring you can use python's interactive prompt. Follow these steps:
```
cd tests\scripts
python
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>> import generate_test_code
>>> help(generate_test_code)
Help on module generate_test_code:

NAME
    generate_test_code

FILE
    c:\users\azikha01\code\github\mazimkhan\mbedtls\tests\scripts\generate_test_code.py

DESCRIPTION
    This script is a key part of Mbed TLS test suites framework. For
    understanding the script it is important to understand the
    framework. This doc string contains a summary of the framework
    and explains the function of this script.

    Mbed TLS test suites:
    =====================
    Scope:
    ------
    The test suites focus on unit testing the crypto primitives and also
    include x509 parser tests. Tests can be added to test any Mbed TLS
    module. However, the framework is not capable of testing SSL
    protocol, since that requires full stack execution and that is best
    tested as part of the system test.

    Test case definition:
    ---------------------
    Tests are defined in a test_suite_<module>[.<optional sub module>].data
    file. A test definition contains:
     test name
     optional build macro dependencies
     test function
     test parameters

    Test dependencies are build macros that can be specified to indicate
    the build config in which the test is valid. For example if a test
    depends on a feature that is only enabled by defining a macro. Then
    that macro should be specified as a dependency of the test.

    Test function is the function that implements the test steps. This
```